src/toil/wdl/wdl_synthesis.py

```diff
@@ -730,7 +733,7 @@ def write_function_cmdline(self, job):
         if 'raw_commandline' in self.tasks_dictionary[job]:
             for cmd in self.tasks_dictionary[job]['raw_commandline']:
                 if not cmd.startswith("r'''"):
-                    cmd = 'str({}).strip("\\n")'.format(cmd)
+                    cmd = 'str({i} if not isinstance({i}, tuple) else process_and_read_file({i}, tempDir, fileStore)).strip("\\n")'.format(i=cmd)
```

Comment:
Oops, I didn't catch this the first time, and maybe I'm mistaken about your intention here, but I don't think this `.strip("\\n")` is doing what you want.

It looks to me like you are trying to get rid of trailing and leading `\\n`s: that is, not newline characters, but the actual characters `\` and `n`. This does that, but it *also* gets rid of other characters if they happen to be either `\`s or `n`s! The strip method works on the basis of characters, not substrings.

OK, that is confusing to try to explain, but an example is worth a thousand words:
```
>>> "\\nnn\\nno correction\\n".strip("\\n")
'o correctio'
```
Couldn't that cause major issues if a command starts/ends with an `n`?

Reply:
Yes, thanks for the catch!  I tested this and yes, misunderstood the logic behind strip().  I did still want to strip normal newlines as well, so I've added a function to perform this and a short test.

I need to test with Cromwell at some point to make sure that Toil's parsing and Cromwell's is one to one, but this seems to work in the meantime.

Reply:
Wait, maybe the double-escaping (because of the indirection through 2 layers of Python) threw me off here.

If I do this:
```
print 'str(cmd if not isinstance(cmd, tuple) else process_and_read_file(cmd, tempDir, fileStore)).strip("\\n")'
```
I get this:
```
str(cmd if not isinstance(cmd, tuple) else process_and_read_file(cmd, tempDir, fileStore)).strip("\n")
```
Which probably means that the original code *is* what you want. Sorry if I led you on a wild goose chase with that one.